spec/index.bs

```diff
@@ -114,33 +168,73 @@ interface GPUAdapter : GPUObjectBase {
 };
 </script>
 
+{{GPUAdapter}} has the following attributes:
+
+<dl dfn-type=attribute dfn-for=GPUAdapter>
+    : <dfn>\[[adapter]]</dfn>, of type [=adapter=], readonly
+    ::
+        An internal slot holding the [=adapter=] to which this {{GPUAdapter}}
+        refers.
+
+    : <dfn>name</dfn>
+    ::
+        A human-readable name identifying the adapter.
+        The contents are implementation-defined.
+
+    : <dfn>extensions</dfn>
+    ::
+        A {{GPUExtensions}} object which enumerates the extensions supported
+        by the user agent, and whether each extension is supported by the
+        underlying implementation.
+          - If an extension is not supported by the user agent,
+            it will not be present in the object.
+          - If an extension is supported by the user agent, but
```

Comment:
It would be nice to leave the possibility for user agents to lie about supported extensions, for example to reduce fingerprinting surface in untrusted contexts.

Reply:
Maybe we could just say if an extensions is *exposed* by the user-agent and supported by the adapter then it will be true, otherwise undefined?

Reply:
There’s “lying” and then there’s “lying.” We can make “lying” implementations intentionally diverge from the spec, or we can make the flex more flexible to allow for intentional “lying.” We should do the latter.

Reply:
To clarify, do you think the spec as written is insufficiently flexible to allow lying? Or just responding to Rafael's comment?

The phrasing "supported by the [=adapter=]" does not yet exist. That is where we would say that the browser can create adapters which "support" whatever they want.